src/Symfony/Component/Security/Core/Tests/Authorization/Voter/AuthenticatedVoterTest.php

```diff
@@ -85,6 +87,29 @@ public function testSupportsType()
         $this->assertTrue($voter->supportsType(get_debug_type(new \stdClass())));
     }
 
+    /**
+     * @dataProvider unsupportedOfflineAttributeProvider
+     */
+    public function testOfflineToken(string $attribute)
```

Comment:
We should add 2 other tests for offline tokens besides the test covering the unsupported case:

- 1 test for `PUBLIC_ACCESS` being granted
- 1 test for an unsupported attribute being abstained.

Reply:
Added.